### DISCUSSION TITLE
Sensor data for plans not updated

### DISCUSSION THREAD

**Discussion by nisnu** [marq24/ha-evcc#20] [Q&A]
Hi,
great integration I have been looking for a while.
In my case the sensor data for the plan are not showing the correct data (evcc_garage_vehicle_plans_soc, evcc_garage_vehicle_plans_time), even if a plan is active in EVCC. I saw in the screenshots of the documentation that they are on state "unknown" as well. Same for me, no matter what is planned.
Has anyone an idea for that behavior?
Thanks.

**Comment by marq24** [comment]
once you have updated to 2024.9.2...

you must enable a plan to be active in order that both of the mentioned entities will show data
![image](https://github.com/user-attachments/assets/c0919218-9db8-4f51-be99-c132e16a1700)

**Comment by h3llrais3r** [comment]
Hi, thanks for this nice integration.
Would it be possible to toggle the charging plan on and off via this intergration?
Because EVCC is currently not having the possibility to limit the charging between 2 hours. (I only want to charge between 21h and 6h).
I've created a plan for departure with date at 6h, but it would start charging my car from the moment I plug it in.
If I'm able to disable/enable the plan from home assistant, I could write an automation to enable the plan at 21h, so it would only start at that hour.

**Reply by marq24** [reply]
I don't see any call in the API in order to disable a plan...
https://docs.evcc.io/docs/integrations/rest-api#vehicles - IMHO only a POST (to create) and a DELETE is available

**Reply by h3llrais3r** [reply]
I see 'plan' api's on both vehicle and loadpoints endpoints.
Do you know what's the difference between them?
![image](https://github.com/user-attachments/assets/dd205796-7d49-40ea-b898-b213fc64cf4e)
![image](https://github.com/user-attachments/assets/5fdac554-4eac-46b4-8a21-25d9b0424810)

Both DELETE actions seems to talk about disabling a plan? Or is it actually deleting the plan?
Vehicle endpoint seems to end with `/soc` while loadpoints endpoint seems to end with `/energy`. 
Behavior is then probably different, but I'm not sure what's the difference...
(I also only find 1 place in UI where you can define a plan, and that's for the vehicle... not for the loadpoint, unless I don't know where it's hidden)

**Comment by UDicke** [comment]
@h3llrais3r I think you can solve this in evcc.
The plan with departure time works for me.
i.e.
![grafik](https://github.com/user-attachments/assets/7bdf9421-eb9d-4e4a-ac46-fc24ff0df90b)

Did you configure a vehicle? evcc has to know how much energie at which power will be needed.
Which tariffs do you use? If you do not use dynamic tarrifs I suggest this feature:
![grafik](https://github.com/user-attachments/assets/74da18d8-481e-467f-bc64-ba218bc3e989)
see here: [https://docs.evcc.io/en/docs/devices/tariffs](url)

**Comment by UDicke** [comment]
> (I also only find 1 place in UI where you can define a plan, and that's for the vehicle... not for the loadpoint, unless I don't know where it's hidden)

I guess loadpoint is used when there is "no vehicle". See this example:
![grafik](https://github.com/user-attachments/assets/4d93e34e-2b06-4e12-99ef-ece465664c67)

![grafik](https://github.com/user-attachments/assets/ef06619d-c9b6-4f87-b219-747dfcfa44e8)

**Comment by h3llrais3r** [comment]
@UDicke, I've setup a car and a plan with departure at 6am.
However, this doesn't prevent my car from being charged from the moment I plug it in.
I only want to start charging at a specific time (9pm).

**Reply by UDicke** [reply]
How is your configuration displayed in evcc?

**Reply by UDicke** [reply]
If the plan looks good and loading starts earlier than the plan shows, there is something wrong with the wallbox or loadpoint.

**Reply by h3llrais3r** [reply]
![image](https://github.com/user-attachments/assets/a089690a-e9dc-41df-98a3-24f7b2301223)

As you can see, if I would plug in my car, it would start charging already.
I really want to hard limit the charging during a defined time window.
If the car doesn't reach the final soc at the end time, it doesn't really matter to me.

**Reply by UDicke** [reply]
To shorten the loadtime simply provide more power than 3,7 kW. Is this possible?

**Reply by UDicke** [reply]
Currently you need over 18h to load
![image](https://github.com/user-attachments/assets/dc744917-0d03-417c-af96-5e1814063ac6)

**Reply by UDicke** [reply]
And if this is not possible you can simply set the pricelimit via HA automation. This works fine with this Integration. I did so for my home battery.
Or you use the tariff option as mentioned above.

**Reply by h3llrais3r** [reply]
Increasing power is possible, but then the windows will change depending on the charge.
I really want hard limits on the window when it can charge. I can then indeed adapt the power if I want to reach the desired percentage.

So setting a priceLimit will block the car from being charged?
So basically I should define a tariff with a price.
![image](https://github.com/user-attachments/assets/337739b4-d682-4e3b-9f27-46c7b6acc5e1)
When setting the price limit to ´0.30´, the plan will only starts during the time that the price is below `0.30`?
So in this case, it will start loading by default, except during the configured zone (mo-fr between 6am - 9pm), as the price is more than ´0.30´?

I suppose this is the api you are talking about:
![image](https://github.com/user-attachments/assets/86181205-854b-4213-b43d-408d11d0f187)
And this sensor:
![image](https://github.com/user-attachments/assets/7850ae7c-0154-4031-8f29-e1d4bb63a315)

**Reply by UDicke** [reply]
I think you should configure like you did above and set the pricelimit to 0,31 Eur/kWh. This should work without further actions.
I don't know whether evcc will round up your 0.2997 to 0,30, just try.
![grafik](https://github.com/user-attachments/assets/9d45c5b9-a13d-436b-ab58-b4fbbe3200a3)
![grafik](https://github.com/user-attachments/assets/26687ab6-8a29-48cb-b016-e32eb53150c9)
I think this will meet your needs without any further actions in HA, just use the functionality of evcc. 😊
The sensor number.evcc_garage_smart_cost_limit would work to set the value, but there is no need to modify it in HA.
In addition I guess you will see the configured tariffs here where I see my actual tibber prices:
![grafik](https://github.com/user-attachments/assets/22ea48ab-c70a-47b7-bdf8-9c514b162da4)

**Reply by h3llrais3r** [reply]
Thanks for the feedback.I really appreciate it.
I've been testing it and it works like you explained above.
I've configured the tariffs as in screenshot I shared in previous comment.
Then I've configured it with a price below 31ct/kWh
![image](https://github.com/user-attachments/assets/b5fe7456-8860-4b01-88c0-1ba4b7a5173e)
It shows that it only will charge between 9pm and 6am. 👍 
If I use this in combination with a plan, which also has a departure at 6am, it plans when it need to start charging to reach the limit of 80% at 6am.
![image](https://github.com/user-attachments/assets/ac3970a5-316d-4292-a9ae-0e66a27120e6)
The plan itself is probably not really needed for the moment, as it just postpones the charging until the time it needs to start to reach the goal at 6am. But going to try it out anyway. 😉 
If I remove the plan, it will charge between 9pm and 6am and stop when 6am is reached or the defined soc on the car is reached, right?

**Reply by UDicke** [reply]
👍 

> If I remove the plan, it will charge between 9pm and 6am and stop when 6am is reached or the defined soc on the car is reached, right?

That´s what I would expect too, I am just a normal user as you are, except that I gave some informations to andig (evcc) to implement the extended batterie mode for saj-h2.

**Reply by h3llrais3r** [reply]
Great, thanks for confirming.
I'm trying to do the same for the saj-h1, but I'm struggling with the batterymode. 
Offtopic: I'm not sure if I'm able to write to the registers via tcpip modbus... I read somewhere that only read would work, and I don't have  intermediate device for `rs485` interface. If you have some info related to this, maybe we can continue in my PR for the saj h1 😉
